src/sentry/http.py

```diff
@@ -183,3 +221,139 @@ def safe_urlopen(url, method=None, params=None, data=None, json=None,
+                timeout=settings.SENTRY_FETCH_SOCKET_TIMEOUT,
+                stream=True,
+            )
+
+            try:
+                cl = int(response.headers['content-length'])
+            except (LookupError, ValueError):
+                cl = 0
+            if cl > settings.SENTRY_FETCH_MAX_SIZE:
+                raise OverflowError()
+
+            contents = []
+            cl = 0
+
+            # Only need to even attempt to read the response body if we
+            # got a 200 OK
+            if response.status_code == 200:
+                for chunk in response.iter_content(16 * 1024):
+                    if time.time() - start > settings.SENTRY_FETCH_TIMEOUT:
+                        raise Timeout()
+                    # we write to a file instead of writing a string
+                    if outfile is not None:
+                        outfile.write(chunk)
+                    else:
+                        contents.append(chunk)
+                    cl += len(chunk)
+                    if cl > settings.SENTRY_FETCH_MAX_SIZE:
+                        raise OverflowError()
+
+        except Exception as exc:
+            logger.debug('Unable to fetch %r', url, exc_info=True)
+            if isinstance(exc, RestrictedIPAddress):
+                error = {
+                    'type': EventError.RESTRICTED_IP,
+                    'url': expose_url(url),
+                }
+            elif isinstance(exc, SuspiciousOperation):
+                error = {
+                    'type': EventError.SECURITY_VIOLATION,
+                    'url': expose_url(url),
+                }
+            elif isinstance(exc, (Timeout, ReadTimeout)):
+                error = {
+                    'type': EventError.FETCH_TIMEOUT,
+                    'url': expose_url(url),
+                    'timeout': settings.SENTRY_FETCH_TIMEOUT,
+                }
+            elif isinstance(exc, OverflowError):
+                error = {
+                    'type': EventError.FETCH_TOO_LARGE,
+                    'url': expose_url(url),
+                    # We want size in megabytes to format nicely
+                    'max_size': float(settings.SENTRY_FETCH_MAX_SIZE) / 1024 / 1024,
+                }
+            elif isinstance(exc, (RequestException, ZeroReturnError)):
+                error = {
+                    'type': EventError.FETCH_GENERIC_ERROR,
+                    'value': six.text_type(type(exc)),
+                    'url': expose_url(url),
+                }
+            else:
+                logger.exception(six.text_type(exc))
+                error = {
+                    'type': EventError.UNKNOWN_ERROR,
+                    'url': expose_url(url),
+                }
+
+            # TODO(dcramer): we want to be less aggressive on disabling domains
+            if domain_lock_enabled:
+                cache.set(domain_key, error or '', 300)
+            logger.warning('source.disabled', extra=error)
```

Comment:
I can happen that a request to iTunes Connect errors out (Session expired or random apple stuff). So we don't want that the iTunes Connect domain gets locked out, we set this to false in the task. But for external sourcemaps that come from the user I guess we like to have this functionality. 
https://github.com/getsentry/sentry-plugins/pull/135/files#diff-ac0402f57ea3be9496e2faa73c4d94b7R119
Fixed the logger though.